android/src/main/java/com/swmansion/gesturehandler/core/FlingGestureHandler.kt

```diff
@@ -35,26 +34,74 @@ class FlingGestureHandler : GestureHandler<FlingGestureHandler>() {
     handler!!.postDelayed(failDelayed, maxDurationMs)
   }
 
-  private fun tryEndFling(event: MotionEvent) = if (
-    maxNumberOfPointersSimultaneously == numberOfPointersRequired &&
-    (
-      direction and DIRECTION_RIGHT != 0 &&
-        event.rawX - startX > minAcceptableDelta ||
-        direction and DIRECTION_LEFT != 0 &&
-        startX - event.rawX > minAcceptableDelta ||
-        direction and DIRECTION_UP != 0 &&
-        startY - event.rawY > minAcceptableDelta ||
-        direction and DIRECTION_DOWN != 0 &&
-        event.rawY - startY > minAcceptableDelta
+  private fun tryEndFling(event: MotionEvent): Boolean {
+    data class SimpleVector(val x: Double, val y: Double)
+
+    fun toSafeNumber(unsafe: Double): Double = if (unsafe.isFinite()) unsafe else 0.0
+
+    fun toUnitVector(vec: SimpleVector): SimpleVector {
+      val magnitude = hypot(vec.x, vec.y)
+      // toSafeNumber protects against division by zero
+      return SimpleVector(
+        toSafeNumber(vec.x / magnitude),
+        toSafeNumber(vec.y / magnitude)
       )
-  ) {
-    handler!!.removeCallbacksAndMessages(null)
-    activate()
-    true
-  } else {
-    false
-  }
+    }
+
+    fun compareSimilarity(
```

Comment:
```suggestion
    fun calculateSimilarity(
```
Given that `compareAlignment` returns `Boolean` and this one returns `Double` I'd say that names should reflect that.